src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicMatch.cs

```diff
@@ -13,17 +13,21 @@ namespace System.Text.RegularExpressions.Symbolic
         /// <summary>Indicates a match was found but without meaningful details about where.</summary>
         internal static SymbolicMatch QuickMatch => new SymbolicMatch(0, 0);
 
-        public SymbolicMatch(int index, int length)
+        public SymbolicMatch(int index, int length, int[]? captureStarts = null, int[]? captureEnds = null)
         {
             Index = index;
             Length = length;
+            CaptureStarts = captureStarts;
+            CaptureEnds = captureEnds;
         }
 
         public int Index { get; }
 
         public int Length { get; }
 
         public bool Success => Index >= 0;
+        public readonly int[]? CaptureStarts;
+        public readonly int[]? CaptureEnds;
```

Comment:
NIT: can we add a brief xml doc fore these two fields? (Even though it does seem obvious what they are)